### DISCUSSION TITLE
HTTP, Color

### DISCUSSION THREAD

**Discussion by Hffhcddrh** [goecharger/go-eCharger-API-v2#251] [Q&A] [answered]
Hallo
ich möchte die farben der LEDs über die HTTP API anpassen.
Leider gelingt mir das nicht. Ich bekomme immer folgende Response

http://192.168.2.213/api/set?cid="0000FF"     --> "{"cid":"cid: failed: invalid color 0000FF (0000FF)"}"
http://192.168.2.213/api/set?cid="#0000FF"  --> "{"cid":"cid: tried to set api key with invalid json IncompleteInput (\")"}"

Was mache ich hier falsch? Hat das jemand schon mal getestet?

**Comment by 0xFEEDC0DE64** [comment]
letztere url schaut schon richtig aus, geht aber vermutlich durch einen browser, der ja blöderweise alles nach dem # als anker nutzt, um zb direkt zu einer gewünschten überschrift zu springen.

Versuch doch mal den http request mit `curl -v 'http://192.168.2.213/api/set?cid="#0000FF"'` abzusetzen, was kommt da raus?

**Comment by Hffhcddrh** [comment, marked as answer]
Hallo,
danke für den Gedankenanstoß mit der #
Mit http://192.168.2.213/api/set?cid="%230000FF" funktioniert es

**Reply by 0xFEEDC0DE64** [reply]
auf's url encoding bin ich gar nicht gekommen, sehr gute lösung!
